internal/bminventory/inventory.go

```diff
@@ -1911,9 +1971,80 @@ func (b *bareMetalInventory) validateAndUpdateClusterParams(ctx context.Context,
 		return installer.V2UpdateClusterParams{}, err
 	}
 
+	if params.ClusterUpdateParams.ControlPlaneCount != nil {
+		if err := validateHighAvailabilityWithControlPlaneCount(
+			swag.StringValue(cluster.HighAvailabilityMode),
+			swag.Int64Value(params.ClusterUpdateParams.ControlPlaneCount),
+			cluster.OpenshiftVersion,
+		); err != nil {
+			return installer.V2UpdateClusterParams{}, err
+		}
+	}
+
 	return *params, nil
 }
 
+func validateHighAvailabilityWithControlPlaneCount(highAvailabilityMode string, controlPlaneCount int64, openshiftVersion string) error {
+	validControlPlaceCountValues := []int64{1, 3, 4, 5}
+	// Updating to invalid control plane count value
+	if !lo.Contains(validControlPlaceCountValues, controlPlaneCount) {
```

Comment:
can we use builtin function instead  ?
```suggestion
	if !slices.Contains(validControlPlaceCountValues, controlPlaneCount) {
```

Reply:
Sure, but we are using `lo` and `funk` everywhere, so I guess it won't matter much